### PR TITLE
KIALI-2529 Fix values that can be undefined or null

### PR DESCRIPTION
**Pull request by cfcosta**
Parameters that come from the URL and from redux can be null or
undefined without causing compilation errors. This does not compile
anymore with `react-scripts`, so it needs to be fixed before the
upgrade.

### CODE REVIEW

File: src/components/MetricsOptions/MetricsSettings.tsx [hunk 1]
```diff
@@ -27,8 +27,8 @@ const secondLevelStyle = style({ marginLeft: 14 });
 const spacerStyle = style({ height: '1em' });
 
 export class MetricsSettingsDropdown extends React.Component<Props> {
-  private shouldReportOptions: boolean;
-  private settings: MetricsSettings;
+  private shouldReportOptions: boolean = false;
```
[the remark this answers is not in the document]
Comment (cfcosta) [reply src/components/MetricsOptions/MetricsSettings.tsx]: They shouldn't. `false !== undefined`.

[the remark this answers is not in the document]
Comment (mtho11) [reply src/components/MetricsOptions/MetricsSettings.tsx]: There is no need to type it when you set the default value. `private shouldReportOptions = false` is enough to establish the type.

**Comment by cfcosta** [comment]
@israel-hdez added your feedback, let's wait for tests and then can you take a look at it again?

**Comment by israel-hdez** [comment]
@cfcosta Tests are green :)
I'll check it again.

File: src/components/Nav/Masthead/UserDropdown.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ class UserDropdownConnected extends React.Component<UserProps, UserState> {
   }
 
   timeLeft = (): number => {
-    const expiresOn = moment(this.props.session.expiresOn);
+    const expiresOn = moment(this.props.session!.expiresOn);
```
[the remark this answers is not in the document]
Comment (israel-hdez) [reply src/components/Nav/Masthead/UserDropdown.tsx]: Session is set even with anonymous strategy. So, no issue.

**Comment by cfcosta** [comment]
@mtho11 @josejulio @jotak what do you guys think? I think we can get this on after the next TP release, as it shouldn't have any problems.

**Comment by cfcosta** [comment]
@mtho11 now that we have branched GA, can we merge this?

**Comment by jshaughn** [comment]
I don't see why not.

**Pull request merged**